utils/testtools/crypto.go

```diff
@@ -18,189 +18,140 @@
 
-// CreateCSR creates CSR.
-func CreateCSR(key crypto.PrivateKey) (csr []byte, err error) {
-	csrDER, err := x509.CreateCertificateRequest(rand.Reader, &x509.CertificateRequest{}, key)
+func NewCertManager() (manager *CertManager, err error) {
+	// Create CA certificate and key
+	ca := &x509.Certificate{
+		SerialNumber: big.NewInt(1658), // nolint:gomnd
+		Subject: pkix.Name{
+			Organization:  []string{"EPAM"},
+			Country:       []string{"BY"},
+			Province:      []string{"Some State"},
+			Locality:      []string{"Middle of the nowhere"},
+			StreetAddress: []string{"You shouldn't know"},
+			PostalCode:    []string{"911"},
+		},
+		DNSNames: []string{
+			"localhost",
+			"wwwivi",
+			"*.westeurope.cloudapp.azure.com",
+			"*.kyiv.epam.com",
+			"*.minsk.epam.com",
+			"*.aos-dev.test",
+			"rabbitmq",
+		},
+		IPAddresses:  []net.IP{net.IPv4(127, 0, 0, 1).To4(), net.IPv6loopback}, // nolint:gomnd
+		NotBefore:    time.Now(),
+		NotAfter:     time.Now().AddDate(10, 0, 0),
+		SubjectKeyId: []byte{1, 2, 3, 4},
+		ExtKeyUsage:  []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:     x509.KeyUsageDigitalSignature,
+	}
+
+	caPrivKey, err := rsa.GenerateKey(rand.Reader, 4096) // nolint:gomnd
 	if err != nil {
 		return nil, aoserrors.Wrap(err)
 	}
 
-	csr = pem.EncodeToMemory(&pem.Block{Type: cryptutils.PEMBlockCertificateRequest, Bytes: csrDER})
-
-	return csr, nil
-}
+	caBytes, err := x509.CreateCertificate(rand.Reader, ca, ca, &caPrivKey.PublicKey, caPrivKey)
+	if err != nil {
+		return nil, aoserrors.Wrap(err)
+	}
 
-// GetCACertificate returns CA certificate.
-func GetCACertificate() (cert []byte) {
-	return []byte(caCertificate)
-}
+	manager = &CertManager{
+		caCert:    ca,
+		caPrivKey: caPrivKey,
+		caCrt:     caBytes,
+	}
 
-// GetCAKey returns CA key.
-func GetCAKey() (cert []byte) {
-	return []byte(caKey)
+	return manager, nil
 }
 
-// CreateCertificate creates certificate.
-func CreateCertificate(workDir string, csr []byte) (cert []byte, err error) {
-	tmpDir, err := ioutil.TempDir(workDir, "cert*")
+// CreateCSR creates CSR.
+func CreateCSR(key crypto.PrivateKey) (csr []byte, err error) {
+	csrDER, err := x509.CreateCertificateRequest(rand.Reader, &x509.CertificateRequest{}, key)
 	if err != nil {
 		return nil, aoserrors.Wrap(err)
 	}
 
-	defer os.RemoveAll(tmpDir)
+	csr = pem.EncodeToMemory(&pem.Block{Type: cryptutils.PEMBlockCertificateRequest, Bytes: csrDER})
 
-	caCertFile := path.Join(tmpDir, "ca.crt")
-	caKeyFile := path.Join(tmpDir, "ca.key")
-	csrFile := path.Join(tmpDir, "unit.csr")
-	csrConfFile := path.Join(tmpDir, "csr.conf")
-	unitCertFile := path.Join(tmpDir, "unit.der")
+	return csr, nil
+}
 
-	if err = ioutil.WriteFile(csrFile, csr, filePerm); err != nil {
-		return nil, aoserrors.Wrap(err)
-	}
+// GetCACertificate returns CA certificate in PEM.
+func (manager *CertManager) GetCACertificate() (cert []byte, err error) {
```

Comment:
in this case you could use not named returns

Reply:
Done

Reply:
not done

Reply:
This method has been removed